src/python/pants/backend/helm/target_types.py

```diff
@@ -465,6 +469,34 @@ class HelmDeploymentFieldSet(FieldSet):
     dependencies: HelmDeploymentDependenciesField
     values: HelmDeploymentValuesField
 
+    def format_values(
+        self, interpolation_context: InterpolationContext, *, ignore_missing: bool = False
+    ) -> dict[str, str]:
+        def format_value(text: str) -> str | None:
+            source = InterpolationContext.TextSource(
+                self.address,
+                target_alias=HelmDeploymentTarget.alias,
+                field_alias=HelmDeploymentValuesField.alias,
+            )
```

Comment:
as the `source` does not depend on the function arg `text`, you could initiate it outside the `format_value()` function body once, rather than for each invocation.